ledger-api/grpc-definitions/com/digitalasset/admin/api/v1/package_management_service.proto

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.digitalasset.admin.api.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option java_outer_classname = "PackageManagementServiceOuterClass";
+option java_package = "com.digitalasset.admin.api.v1";
+
+// Allows clients to query the DAML-LF packages that are supported by the ledger.
+service PackageManagementService {
+
+  // Returns the details of all supported packages.
+  rpc ListPackageDetails (ListPackageDetailsRequest) returns (ListPackageDetailsResponse);
+
+  // Uploads a DAR archive.
+  rpc UploadDar (UploadDarRequest) returns (UploadDarResponse);
+
+}
+
+message ListPackageDetailsRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message ListPackageDetailsResponse {
+
+  // The IDs of all DAML-LF packages supported by the ledger.
+  // Required
+  repeated string PackageDetails = 1;
+}
+
+message PackageDetails {
+ 
+  // The IDs of all DAML-LF packages supported by the ledger.
```

Comment:
i think this description is out of place. it says "The IDs of all DAML-LF packages..." to describe one field.